pkg/kubelet/nodeconfig/rollback.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeconfig
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	ccv1a1 "k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
+)
+
+// badRollback makes an entry in the bad-config-tracking file for `uid` with `reason`, and then
+// returns the result of rolling back to the last-known-good config.
+// If filesystem issues prevent marking the config bad or rolling back, a fatal error occurs.
+func (cc *NodeConfigController) badRollback(uid, reason, detail string) *ccv1a1.KubeletConfiguration {
+	if len(detail) > 0 {
+		detail = fmt.Sprintf(", %s", detail)
+	}
+	errorf(fmt.Sprintf("%s%s", reason, detail))
```

Comment:
Agreed, we should always try to call it with a detail. But the reason should be separate. It gets reported in a node status, and we don't want to bloat those messages with arbitrary, possibly low-level errors that actually belong in a log.

As far as not having to check the length of detail, I agree that's spurious.